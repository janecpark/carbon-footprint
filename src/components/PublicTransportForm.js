import React from 'react';
import { useHistory } from 'react-router-dom';
import bus from '../images/bus.jpg';

const PublicTransportForm = () => {
  const history = useHistory();
  const handleChange = () => {};
  const handleSubmit = (e) => {
    e.preventDefault();
    history.push('/home1');
  };

  const style = {
    backgroundImage: `url('${bus}`,
    height: '100vh',
    backgroundRepeat: 'no-repeat',
    backgroundPosition: 'center',
    backgroundSize: 'cover',
    width: '100vw',
  };

  return (
    <div className="PublicTransportForm" >
      <form onSubmit={handleSubmit} className="container">
        <div className="form-group">
          <label htmlFor="public-transit">Public Transit</label>
          <div className="input-group">
            <input
              type="text"
              className="form-control"
              id="public-transit"
              placeholder="413"
            ></input>
            <label className="input-group-text">mi/yr</label>
            <div className="input-group-addon"></div>
          </div>
        </div>
        <div className="form-group">
          <label htmlFor="air-transit">Air Travel</label>
          <div className="input-group">
            <input
              type="text"
              className="form-control"
              placeholder="3,700"
              id="air-transit"
            ></input>
            <label className="input-group-text">mi/yr</label>
            <div className="input-group-addon"></div>
          </div>
        </div>
        {/* <button className="btn btn-success mr-2">Previous</button>
        <button className="btn btn-success">Next</button> */}
        <button className="btn btn-success">Next</button>
      </form>
    </div>
  );
};

export default PublicTransportForm;
