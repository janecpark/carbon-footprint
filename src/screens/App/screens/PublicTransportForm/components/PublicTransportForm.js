import React, { useContext, useState } from 'react';
import { useHistory } from 'react-router-dom';
import FormContext from '../../../FormContext';

/** Shows a form to get information about public transportation  */

const PublicTransportForm = () => {
  const history = useHistory();
  const [transport, setTransport] = useState();
  const { setFormData } = useContext(FormContext);
  const handleChange = (e) => {
    setTransport(e.target.value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    setFormData((prevState) => ({
      ...prevState,
      input_footprint_transportation_publictrans: transport || 452,
    }));
    history.push('/home1');
  };

  return (
    <div className="PublicTransportForm">
      <form onSubmit={handleSubmit} className="container">
        <div className="form-group">
          <h4>Public Transit</h4>
          <div className="input-group">
            <input
              type="text"
              shadow-none
              className="form-control shadow-none"
              id="public-transit"
              placeholder="413 (optional)"
              onChange={handleChange}
              autoFocus
            ></input>
            <label className="input-group-text">mi/yr</label>
            <div className="input-group-addon"></div>
          </div>
        </div>
        <button className="btn btn-success">Next</button>
      </form>
    </div>
  );
};

export default PublicTransportForm;
