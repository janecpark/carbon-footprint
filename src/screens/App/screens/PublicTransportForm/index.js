import PublicTransportForm from './components/PublicTransportForm';

export default PublicTransportForm;