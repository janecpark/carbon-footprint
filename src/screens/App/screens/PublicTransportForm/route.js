import { routeConstants } from '../../../../shared/constants';
import PublicTransportForm from "./";

export default {
    path: routeConstants.PUBLICTRANSPORT.route,
    component: PublicTransportForm
};