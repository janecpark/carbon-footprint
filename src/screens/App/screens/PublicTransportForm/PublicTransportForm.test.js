import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import PublicTransportForm from './components/PublicTransportForm';
import {
  UserProvider,
  FormProvider,
  ResultProvider,
  ActionProvider,
} from '../testUtils';

it('renders without crashing', function () {
  render(
    <MemoryRouter>
      <UserProvider>
        <FormProvider>
          <ResultProvider>
            <ActionProvider>
              <PublicTransportForm />
            </ActionProvider>
          </ResultProvider>
        </FormProvider>
      </UserProvider>
    </MemoryRouter>
  );
});

it('matches snapshot', function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <FormProvider>
          <ResultProvider>
            <ActionProvider>
              <PublicTransportForm />
            </ActionProvider>
          </ResultProvider>
        </FormProvider>
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

test('renders text on page', () => {
  const { getByText } = render(
    <MemoryRouter>
      <UserProvider>
        <FormProvider>
          <ResultProvider>
            <ActionProvider>
              <PublicTransportForm />
            </ActionProvider>
          </ResultProvider>
        </FormProvider>
      </UserProvider>
    </MemoryRouter>
  );
  const element = getByText(/Next/);
  expect(element).toBeInTheDocument();
});
