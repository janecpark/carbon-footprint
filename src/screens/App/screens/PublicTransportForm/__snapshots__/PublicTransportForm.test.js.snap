// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="PublicTransportForm"
  >
    <form
      class="container"
    >
      <div
        class="form-group"
      >
        <h4>
          Public Transit
        </h4>
        <div
          class="input-group"
        >
          <input
            class="form-control"
            id="public-transit"
            placeholder="413 (optional)"
            type="text"
          />
          <label
            class="input-group-text"
          >
            mi/yr
          </label>
          <div
            class="input-group-addon"
          />
        </div>
      </div>
      <button
        class="btn btn-success"
      >
        Next
      </button>
    </form>
  </div>
</DocumentFragment>
`;
